code/datums/flock/flockprotection.dm

```diff
@@ -1,76 +1,83 @@
+
+/// Mark a specific flock as interested in this
+/datum/component/flock_interest
+	/// The flock who is intently interested in this thing.
+	var/datum/flock/flock
+
+/datum/component/flock_interest/Initialize(datum/flock/flock)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.flock = flock
+
+/datum/component/flock_interest/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_FLOCK_ATTACK, .proc/handle_flock_attack)
+
+/// If flockdrone is in our flock, deny the attack, otherwise scream and cry
+/datum/component/flock_interest/proc/handle_flock_attack(source, mob/user, var/intentional)
+	if(!istype(user))
+		return
+
+	var/mob/living/critter/flock/F = user
+	if (istype(F) && F.flock == src.flock)
+		if(intentional)
+			boutput(user, "<span class='alert'>The grip tool refuses to harm this, jamming briefly.</span>")
+		return intentional
+
+	var/mob/living/critter/flock/drone/snitch
+	for (var/mob/living/critter/flock/drone/FD in viewers(user))
+		if (FD != user && !isdead(FD) && FD.is_npc && FD.flock == src.flock)
```

Comment:
```suggestion
		if (FD != user && !isdead(FD) && FD.is_npc && !FD.dormant && FD.flock == src.flock)
```
not sure why this didn't attach to the review

Reply:
Atm, dormantizing a drone automatically removes it from the flock.